src/centos/scripts/start-marklogic.sh

```diff
@@ -147,6 +147,49 @@ function restart_check {
     error "Failed to restart $1" exit
 }
 
+################################################################
+# setting default protocol and CA cert for curl calls to bootstrap host
+################################################################
+ML_BOOTSTRAP_PROTOCOL="http"
+ML_CACERT_FILE="/run/secrets/${MARKLOGIC_JOIN_CACERT_FILE}"
+
+################################################################
+# Check if TLS is enabled on Bootstrap host
+################################################################
+function validate_tls_parameters {
+    local bootstrap_status=$1 host_protocol
+    host_protocol=$(get_host_protocol "${MARKLOGIC_BOOTSTRAP_HOST}")
+    if [[ "${bootstrap_status}" != "localhost" ]]; then
+        if [[ "${MARKLOGIC_JOIN_TLS_ENABLED}" == "true" ]]; then
+            if [[ -f "${ML_CACERT_FILE}" ]] && [[ -n "$(<"${ML_CACERT_FILE}")" ]]; then
+                if [[ "${host_protocol}" == "https" ]]; then
+                    ML_BOOTSTRAP_PROTOCOL="https"
+                    # validate the CA certificate
+                    validate_cert "${ML_CACERT_FILE}"
```

Comment:
This does not exit when CA is not valid. I think the requirement is to exit the process when CA is invalid?

Reply:
the function validate_cert will exit if the certificate is not valid.